src/javascript/app_2/Stores/Modules/Contract/contract-store.js

```diff
@@ -173,6 +190,20 @@ export default class ContractStore extends BaseStore {
         }
     }
 
+    handleChartType(SmartChartStore, start, expiry) {
+        const chart_type  = getChartType(start, expiry);
+        const granularity = getChartGranularity(start, expiry);
+
+        if (chart_type === 'candle' && granularity !== 0) {
+            SmartChartStore.granularity = granularity;
```

Comment:
Should this use `SmartChartStore. updateGranularity(granularity)`?

and should be same but for `chart_type` also?

Reply:
No, because we're directly updating the `granularity` and `chart_type` props and the `updateGranularity` and `updateChartType` `onChange` functions are only to receive back the props updated  from `SmartCharts`.

Reply:
Ok I was just thinking that usually we don't update stores from the outside directly we go through "setters", shouldn't the outcome be the same either way?

Reply:
yeah i get what you mean, this was the previous approach as well, but it caused issues in previous versions of `smartcharts`, two versions back prior to `0.4.18`. I'll set it back to using the setters for now since it's working fine on `0.4.19`.